apps/projects/contracts/Projects.sol

```diff
@@ -427,32 +420,38 @@ contract Projects is IsContract, AragonApp {
     }
 
     /**
-     * @notice Review work submitted by '`_contributor`'.
+     * @notice Review work submission.
      * @dev add a submission to local state after it's been added to StandardBounties.sol
      * @param _repoId the github repo id of the issue
      * @param _issueNumber the github issue up for resolution
-     * @param _contributor address of the asignee who submitted work for review
-     * @param _approved decision to accept the contribution 
+     * @param _submissionNumber submission index of the submitted work for review
+     * @param _approved decision to accept the contribution
      */
     function reviewSubmission(
-        bytes32 _repoId, 
-        uint256 _issueNumber, 
-        address _contributor,
-        bool _approved
+        bytes32 _repoId,
+        uint256 _issueNumber,
+        uint256 _submissionNumber,
+        bool _approved,
+        string updatedSubmissionHash
     ) external isInitialized auth(WORK_REVIEW_ROLE)
     {
         GithubIssue storage issue = repos[_repoId].issues[_issueNumber];
-        require(!issue.fulfilled,"Bounty already fulfilled");
+
+        require(!issue.fulfilled,"BOUNTY_FULFILLED");
+
+        WorkSubmission storage submission = workSubmissions[issue.submissionIndices[_submissionNumber]];
+        submission.submissionHash = updatedSubmissionHash;
 
         if (_approved) {
             if (issue.hasBounty) {
-                bounties.acceptFulfillment(issue.standardBountyId, issue.workSubmissions[_contributor].fulfillmentId);
+                bounties.acceptFulfillment(issue.standardBountyId, submission.fulfillmentId);
             }
             issue.fulfilled = true;
-            issue.workSubmissions[_contributor].status = SubmissionStatus.Accepted;
-            emit SubmissionAccepted(_contributor, _repoId, _issueNumber);
+            submission.status = SubmissionStatus.Accepted;
+            emit SubmissionAccepted(_submissionNumber, _repoId, _issueNumber);
```

Comment:
Should it worth to just emit a single `SubmissionResult` event with a boolean accepted/rejected param? I don't know if that would improve something in terms of gas, but I imagine it maybe would be easier to handle from frontend? Open to opinions on that

Reply:
I'm going to table this for discussion later